pkg/asset/agent/manifests/common.go

```diff
@@ -22,6 +24,14 @@ func getPullSecretName(ic *agent.OptionalInstallConfig) string {
 	return ic.ClusterName() + "-pull-secret"
 }
 
+func getProxy(ic *agent.OptionalInstallConfig) *aiv1beta1.Proxy {
```

Comment:
I think we should really stop with this approach, for a couple of reasons:

* Defining a method far away from the original type is not maintenance-friendly and detrimental for the readability
* The consumer code should have a reference only to `types.InstallConfig`. The syntethic type `OptionalInstallConfig` should be used only to extract the underlying installConfig if present, and no other responsibility should be given to that type

Reply:
Agreed. I think making a refactor after the branching is a good idea

Reply:
The reason for most of these functions to exist is that they are used in multiple ZTP manifests to produce data that has to match up for the ZTP manifests to work correctly, and their implementations may change in the future. e.g. the PullSecret name has to be the same in the Secret, the InfraEnv, and the ClusterDeployment - hence `getPullSecretName()`.

I don't know at what point this became a cargo-cult thing, but there is absolutely no reason for this function to exist. We can just get the data out of the InstallConfig where we need it.